test/e2e/magic_catalog.go

```diff
@@ -50,74 +52,119 @@ func NewMagicCatalog(kubeClient k8scontrollerclient.Client, namespace string, ca
 }
 
 func (c *magicCatalog) DeployCatalog(ctx context.Context) error {
-
 	catalogSource := c.makeCatalogSource()
 	resourcesInOrderOfDeployment := []k8scontrollerclient.Object{
 		c.makeConfigMap(),
 		c.makeCatalogSourcePod(),
 		c.makeCatalogService(),
 		catalogSource,
 	}
+	if err := c.deployCatalog(ctx, resourcesInOrderOfDeployment); err != nil {
+		return err
+	}
+	if err := catalogSourceIsReady(ctx, c.kubeClient, catalogSource); err != nil {
+		return c.cleanUpAfter(ctx, err)
+	}
+
+	return nil
+}
+
+func catalogSourceIsReady(ctx context.Context, c k8scontrollerclient.Client, cs *operatorsv1alpha1.CatalogSource) error {
+	// wait for catalog source to become ready
+	return waitFor(func() (bool, error) {
+		err := c.Get(ctx, k8scontrollerclient.ObjectKey{
+			Name:      cs.GetName(),
+			Namespace: cs.GetNamespace(),
+		}, cs)
+		if err != nil || cs.Status.GRPCConnectionState == nil {
+			return false, err
+		}
+		state := cs.Status.GRPCConnectionState.LastObservedState
+		if state != catalogReadyState {
+			return false, nil
+		}
+		return true, nil
+	})
```

Comment:
```suggestion
return state == catalogReadyState, nil
```

Reply:
Yep - I had noticed this locally too but forgot to push up a change.